### PR TITLE
[WIP] convert HBN ASR

### PR DESCRIPTION
**Pull request by yibeichan**
I converted ASR (one questionnaire from HBN) as my first try for reproschema. It would be great if @sooyounga or @djarecka can take a look at it (especially `items/ASR_126`). (btw, @djarecka I went through LinkML tutorials but think it's easier to write hard-coded python script for this first converting since I'm not that familiar with schema yet. )

Also, based on my understanding of this questionnaire, our final output should be structured similar to [this one](https://github.com/ReproNim/reproschema-library/tree/master/activities/CIT), where items should be grouped into sub-categories (e.g., ASR_SC, ASR_AB, ASR_WD, etc.). I don't have detailed information about which items should be group which category at this moment but I'll update the structure once I have more info on the questionnaire.

### CODE REVIEW

**Comment by yibeichan** [comment]
for reference, i put my python script here (with directory info removed):

```
df = pd.read_csv('hbn/item-names-cleaned.csv')

# choose one assessment
asr_df = df[df["datadic"]=="ASR"]
asr_df.dropna(subset=["keys"], inplace=True)
sub_asr_df = asr_df[asr_df["keys"].str.match(r'^ASR_\d+[a-z]?$')].copy()
# fill nan as 999
sub_asr_df["value"].fillna(999, inplace=True)

# create json file for each item
for index, row in sub_asr_df.iterrows():
    var = row["keys"]
    dom = row["domains"].replace('_', ' ')
    desc = f"Q{index+1} of the {dom}"
    question = row["questions"].replace(r'^\d+\.\s', '')
    if row["value"] != 999:
        itype = "radio"
        resopt = "../valueConstraints"
    else:
        itype = "text"
        resopt = {
            "valueType": "xsd:string"
            }
    item_json = {
        "@context": "https://raw.githubusercontent.com/ReproNim/reproschema/1.0.0-rc4/contexts/generic",
        "@type": "reproschema:Field",
        "@id": var,
        "prefLabel": var,
        "description": desc,
        "schemaVersion": "1.0.0-rc4",
        "version": "0.0.1",
        "question": {
            "en": question
        },
        "ui": {
            "inputType": itype
        },
        "responseOptions": resopt
    }
    with open(os.path.join(items_dir, f"{var}"), 'w', encoding='utf-8') as jsonfile:
        json.dump(item_json, jsonfile, indent=4)

# create the main json file
vars = sub_asr_df["keys"]
# Build the JSON-LD object for the main file
jsonld_obj = {
    "@context": "https://raw.githubusercontent.com/ReproNim/reproschema/1.0.0-rc1/contexts/generic",
    "@type": "reproschema:Activity",
    "@id": "hbn_asr",
    "prefLabel": "HBN Adult Self Report (ASR)",
    "altLabel": "hbn_asr",
    "schemaVersion": "1.0.0-rc1",
    "version": "0.0.1",
    "ui": {
        "order": [f"items/{var}" for var in vars],
        "shuffle": False,
        "addProperties": [{"variableName": var, "isAbout": f"items/{var}"} for var in vars]
    }
}

# Write the JSON-LD object to a file
with open(os.path.join(output_dir, "hbn_asr_schema"), 'w', encoding='utf-8') as jsonfile:
    json.dump(jsonld_obj, jsonfile, indent=4)
```

**Comment by djarecka** [comment]
`item-names-cleaned.csv` is somewhere online?

File: activities/hbn_asr/hbn_asr_schema [hunk 1]
```diff
@@ -0,0 +1,661 @@
+{
+    "@context": "https://raw.githubusercontent.com/ReproNim/reproschema/1.0.0-rc1/contexts/generic",
+    "@type": "reproschema:Activity",
+    "@id": "hbn_asr",
```
Comment (djarecka) [inline activities/hbn_asr/hbn_asr_schema:4]: seems like we expect this just to be the same as the name of the file

File: activities/hbn_asr/hbn_asr_schema [hunk 2]
```diff
@@ -0,0 +1,661 @@
...
+    "prefLabel": "HBN Adult Self Report (ASR)",
+    "altLabel": "hbn_asr",
+    "schemaVersion": "1.0.0-rc1",
```
Comment (djarecka) [inline activities/hbn_asr/hbn_asr_schema:7]: we have already newer versions, rc4

File: activities/hbn_asr/hbn_asr_schema [hunk 3]
```diff
@@ -0,0 +1,661 @@
...
+            "items/ASR_123",
+            "items/ASR_124",
+            "items/ASR_125",
+            "items/ASR_126"
+        ],
+        "shuffle": false,
+        "addProperties": [
+            {
+                "variableName": "ASR_01",
+                "isAbout": "items/ASR_01"
```
Comment (djarecka) [inline activities/hbn_asr/hbn_asr_schema:145]: I think, we should also have `valueRequired` and `isVis` fields, as in the CIT schema

File: activities/hbn_asr/items/ASR_01 [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
+    "@context": "https://raw.githubusercontent.com/ReproNim/reproschema/1.0.0-rc4/contexts/generic",
+    "@type": "reproschema:Field",
+    "@id": "ASR_01",
+    "prefLabel": "ASR_01",
+    "description": "Q1 of the Questionnaire Measures of Emotional and Cognitive Status",
+    "schemaVersion": "1.0.0-rc4",
+    "version": "0.0.1",
+    "question": {
+        "en": "1. I am too forgetful"
```
Comment (djarecka) [inline activities/hbn_asr/items/ASR_01:10]: I would think that the numbers could be removed from the question itself

File: activities/hbn_asr/items/ASR_126 [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    "@id": "ASR_126",
+    "prefLabel": "ASR_126",
+    "description": "Q129 of the Questionnaire Measures of Emotional and Cognitive Status",
+    "schemaVersion": "1.0.0-rc4",
+    "version": "0.0.1",
+    "question": {
+        "en": "126. In the past 6 months, on how many days did you use drugs for nonmedical purposes (including marijuana, cocaine, and other drugs, except alcohol and nicotine)?"
+    },
+    "ui": {
+        "inputType": "text"
```
Comment (djarecka) [inline activities/hbn_asr/items/ASR_126:13]: i was wondering if this should be a text or some int, not sure of that was specified somewhere

Comment (yibeichan) [reply activities/hbn_asr/items/ASR_126]: so now i changed it to 
```
itype = "number"
        resopt = {
            "valueType": "xsd:integer"
            }
```

File: activities/hbn_asr/hbn_asr_schema (hunk 3 above)
Comment (yibeichan) [reply activities/hbn_asr/hbn_asr_schema]: added

File: activities/hbn_asr/hbn_asr_schema (hunk 2 above)
Comment (yibeichan) [reply activities/hbn_asr/hbn_asr_schema]: updated

**Comment by yibeichan** [comment]
@djarecka can you run the workflow again?

**Pull request merged**